src/Microsoft.Health.Dicom.Api/Registration/DicomServerServiceCollectionExtensions.cs

```diff
@@ -45,12 +48,24 @@ public static class DicomServerServiceCollectionExtensions
     /// Add services for DICOM background workers.
     /// </summary>
     /// <param name="serverBuilder">The DICOM server builder instance.</param>
+    /// <param name="configuration">The configuration for the DICOM server.</param>
     /// <returns>The DICOM server builder instance.</returns>
-    public static IDicomServerBuilder AddBackgroundWorkers(this IDicomServerBuilder serverBuilder)
+    public static IDicomServerBuilder AddBackgroundWorkers(this IDicomServerBuilder serverBuilder, IConfiguration configuration)
     {
         EnsureArg.IsNotNull(serverBuilder, nameof(serverBuilder));
         serverBuilder.Services.AddScoped<DeletedInstanceCleanupWorker>();
         serverBuilder.Services.AddHostedService<DeletedInstanceCleanupBackgroundService>();
+
+        serverBuilder.Services.AddCustomerKeyValidationBackgroundService(options =>
```

Comment:
nit: Let's ditch the code blocks and use expressions. Perhaps something like this:
```csharp
serverBuilder.Services
    .AddCustomerKeyValidationBackgroundService(options => configuration
        .GetSection(CustomerManagedKeyOptions.CustomerManagedKey)
        .Bind(options))
    .AddHealthCheckPublisher(options => configuration
        .GetSection("HealthCheckPublisher")
        .Bind(options));
```